src/api/EscargotPublic.cpp

```diff
@@ -4265,19 +4265,28 @@ WeakSetObjectRef* WeakSetObjectRef::create(ExecutionStateRef* state)
     return toRef(new WeakSetObject(*toImpl(state)));
 }
 
-void WeakSetObjectRef::add(ExecutionStateRef* state, ObjectRef* key)
+void WeakSetObjectRef::add(ExecutionStateRef* state, ValueRef* key)
```

Comment:
If value is not `Heldable`, how about return false?

Reply:
I fixed it to return boolean as you commented 
Thanks for the review!